src/Diagnostics.ModelsAndUtils/Attributes/ResourceFilter/Enumerations.cs

```diff
@@ -0,0 +1,94 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Diagnostics.ModelsAndUtils.Attributes
+{
+    /// <summary>
+    /// Enum representing Resource type or collection of Resource Types
+    /// </summary>
+    public enum ResourceType
+    {
+        /// <summary>
+        /// App Resource
+        /// </summary>
+        App = 1,
+
+        /// <summary>
+        /// Hosting Environment Resource
+        /// </summary>
+        HostingEnvironment = 2
+    }
+
+    /// <summary>
+    /// Enum reprensenting Platform type.
+    /// </summary>
+    public enum PlatformType
+    {
+        /// <summary>
+        /// Reprensents windows platform
+        /// </summary>
+        Windows = 1,
+
+        /// <summary>
+        /// Represents linux platform
+        /// </summary>
+        Linux = 2
+    }
+
+    /// <summary>
+    /// Represents different App Types
+    /// </summary>
+    public enum AppType
+    {
+        WebApp = 1,
+        FunctionApp = 2,
+        ApiApp = 4,
+        MobileApp = 8,
+        LogicApp = 16,
+        GatewayApp = 32,
+        All = 255
+    }
+
+    /// <summary>
+    /// Hosting Environment Types
+    /// </summary>
+    public enum HostingEnvironmentType
+    {
+        None = 0,
+        V1 = 1,
+        V2 = 2,
+        All = 255
+    }
+
+    public enum StackType
+    {
+        None = 0,
+        AspNet = 1,
+        NetCore = 2,
+        Php = 4,
+        Python = 8,
+        Node = 16,
+        Java = 32,
+        Static = 64,
+        Other = 128,
+        All = 255
+    }
+
+    public enum SkuType
+    {
+        Free = 1,
+        Shared = 2,
+        Standard = 4,
+        Premium = 8,
+        All = 255
+    }
+
+    public enum StampType
+    {
+        Public = 1,
+        ASEV1 = 2,
+        ASEV2 = 4,
+        All = 255
+    }
+}
```

Comment:
Megastamp?

Reply:
you see a scenario applicable only for megastamp. I can add that.

Reply:
I am trying to think why this is needed and now I am realizing most of the times it is for people when forming KUSTO queries so it may not be that needed. Actually rather than an Enum type, it would make sense to have a method that one can call IsMegastamp? I cannot think of real world scenario but sometimes we have specific issues only on Megstamps. Anyways not a high priority.